src/main/java/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,32 @@
+package service.impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        if (transactions == null) {
+            throw new IllegalArgumentException("Transactions cannot be null");
+        }
+
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            if (handler == null) {
+                throw new RuntimeException(
+                        "No handler found for operation: " + transaction.getOperation());
+            }
+            handler.apply(transaction);
```

You call `handler.apply(transaction)` without wrapping exceptions to add context. The task requires: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and original cause." If a handler throws (for example on a purchase that would make stock negative), catch the exception, wrap it in a RuntimeException (or a specific unchecked type) that includes the `transaction` info and the original cause, and rethrow. This makes failures easier to debug and satisfies the error-handling requirement.